src/electionguard_gui/web/components/admin/view-decryption-admin-component.js

```diff
@@ -29,26 +29,37 @@ export default {
         spoiledBallotId: spoiledBallotId,
       });
     },
-    refresh_decryption: async function () {
-      await this.get_decryption(true);
+    refresh_decryption: async function (result) {
+      if (result.success) {
+        await this.get_decryption(true);
+      } else {
+        console.error(result.message);
+        this.error = true;
+      }
     },
     get_decryption: async function (is_refresh) {
       console.log("getting decryption");
       this.loading = true;
-      const result = await eel.get_decryption(this.decryptionId, is_refresh)();
-      this.error = !result.success;
-      if (result.success) {
-        this.decryption = result.result;
+      try {
+        const result = await eel.get_decryption(
+          this.decryptionId,
+          is_refresh
+        )();
+        this.error = !result.success;
+        if (result.success) {
+          this.decryption = result.result;
```

Comment:
on a failure, we probably want to set the this.decryption back to an initial state so that we are not using a different one.

Reply:
An excellent point in general.  In this particular instance, however, setting the decryption to null would make most of the UI disappear because of the v-if on line 74.  I'm not sure we necessarily want that.  It would require the user to either hit the back button (alt-left) or Ctrl+R to reload the app, or to add some UI to handle that scenario.